components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/GatewayPolicyDeployer.java

```diff
@@ -0,0 +1,187 @@
+ * the gateway policy mappings in gateway.
+ */
+public class GatewayPolicyDeployer {
+    private static final Log log = LogFactory.getLog(GatewayPolicyDeployer.class);
+    ArtifactRetriever artifactRetriever;
+    private String gatewayPolicyMappingUuid;
+    GatewayArtifactSynchronizerProperties gatewayArtifactSynchronizerProperties;
+
+    public GatewayPolicyDeployer(String gatewayPolicyMappingUuid) {
+        this.artifactRetriever = ServiceReferenceHolder.getInstance().getArtifactRetriever();
+        this.gatewayPolicyMappingUuid = gatewayPolicyMappingUuid;
+        this.gatewayArtifactSynchronizerProperties = ServiceReferenceHolder
+                .getInstance().getAPIManagerConfiguration().getGatewayArtifactSynchronizerProperties();
+    }
+
+    public GatewayPolicyDeployer() {
+        this.artifactRetriever = ServiceReferenceHolder.getInstance().getArtifactRetriever();
+        this.gatewayArtifactSynchronizerProperties = ServiceReferenceHolder
+                .getInstance().getAPIManagerConfiguration().getGatewayArtifactSynchronizerProperties();
+    }
+
+    /**
+     * Deploy gateway policy in the gateway using the deployGatewayPolicy method in gateway admin.
+     *
+     * @throws ArtifactSynchronizerException
+     */
+    public void deployGatewayPolicyMapping() throws ArtifactSynchronizerException {
+        try {
+            GatewayPolicyDTO gatewayPolicyDTO = retrieveGatewayPolicyArtifact(gatewayPolicyMappingUuid);
+            if (gatewayPolicyDTO != null) {
+                APIGatewayAdmin apiGatewayAdmin = new APIGatewayAdmin();
+                MessageContext.setCurrentMessageContext(
+                        org.wso2.carbon.apimgt.gateway.utils.GatewayUtils.createAxis2MessageContext());
+                apiGatewayAdmin.deployGatewayPolicy(gatewayPolicyDTO);
+            } else {
+                log.error("Gateway policy artifact is not found for the policy mapping: " + gatewayPolicyMappingUuid);
+            }
+        } catch (AxisFault e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * Undeploy gateway policy in the gateway using the unDeployGatewayPolicy method in gateway admin.
+     *
+     * @throws ArtifactSynchronizerException
+     */
+    public void undeployGatewayPolicyMapping() throws ArtifactSynchronizerException {
+        try {
+            GatewayPolicyDTO gatewayPolicyDTO = retrieveGatewayPolicyArtifact(gatewayPolicyMappingUuid);
+            if (gatewayPolicyDTO != null) {
+                APIGatewayAdmin apiGatewayAdmin = new APIGatewayAdmin();
+                MessageContext.setCurrentMessageContext(
+                        org.wso2.carbon.apimgt.gateway.utils.GatewayUtils.createAxis2MessageContext());
+                apiGatewayAdmin.unDeployGatewayPolicy(gatewayPolicyDTO);
+            }
+        } catch (AxisFault e) {
+            throw new RuntimeException(e);
```

Comment:
Same here

Reply:
resolved with https://github.com/wso2/carbon-apimgt/pull/12201/commits/4042971246cf43e60948916f7de0a263bb8f2344

thank you